barretenberg/cpp/src/barretenberg/sumcheck/sumcheck.test.cpp

```diff
@@ -289,15 +289,100 @@ template <typename Flavor> class SumcheckTests : public ::testing::Test {
                 verifier_transcript->template get_challenge<FF>("Sumcheck:gate_challenge_" + std::to_string(idx));
         }
 
-        std::vector<FF> padding_indicator_array(virtual_log_n, 1);
-        if constexpr (Flavor::HasZK) {
-            for (size_t idx = 0; idx < virtual_log_n; idx++) {
-                padding_indicator_array[idx] = (idx < multivariate_d) ? FF{ 1 } : FF{ 0 };
-            }
+        auto verifier_output = sumcheck_verifier.verify(relation_parameters, verifier_gate_challenges);
+
+        auto verified = verifier_output.verified;
+
+        EXPECT_EQ(verified, true);
+    };
+
+    // TODO(#225): make the inputs to this test more interesting, e.g. non-trivial permutations
+    void test_zk_prover_verifier_flow()
+    {
+        const size_t multivariate_d(3);
+        const size_t multivariate_n(1 << multivariate_d);
+
+        const size_t virtual_log_n = 24;
+        // Construct prover polynomials where each is the zero polynomial.
+        // Note: ProverPolynomials are defined as spans so the polynomials they point to need to exist in memory.
+        std::vector<Polynomial<FF>> zero_polynomials(NUM_POLYNOMIALS);
+        for (auto& poly : zero_polynomials) {
+            poly = bb::Polynomial<FF>(multivariate_n);
         }
+        auto full_polynomials = construct_ultra_full_polynomials(zero_polynomials);
 
-        auto verifier_output =
-            sumcheck_verifier.verify(relation_parameters, verifier_gate_challenges, padding_indicator_array);
+        // Add some non-trivial values to certain polynomials so that the arithmetic relation will have non-trivial
+        // contribution. Note: since all other polynomials are set to 0, all other relations are trivially
+        // satisfied.
+        std::array<FF, multivariate_n> w_l = { 0, 0, 0, 7, 0, 1, 2, 0 };
```

Comment:
Can you say something about where these values are coming from? Youre just populating some "random" values in the hiding slots?